src/NuGet.Core/NuGet.Build.Tasks/NuGet.targets

```diff
@@ -139,7 +139,15 @@ Copyright (c) .NET Foundation. All rights reserved.
       HideWarningsAndErrors="$(HideWarningsAndErrors)"
       Interactive="$(NuGetInteractive)"
       RestoreForceEvaluate="$(RestoreForceEvaluate)"
-      RestorePackagesConfig="$(RestorePackagesConfig)"/>
+      RestorePackagesConfig="$(RestorePackagesConfig)">
+      <Output TaskParameter="RestoredProjectOutputPaths" ItemName="_RestoredProjectOutputPaths" />
```

Comment:
Can we return the exact files here instead of paths to do globbing for? We know the exact paths from within the task + we can account for packages.config projects.